src/react/components.js

```diff
@@ -57,6 +57,7 @@ export function getInitialProps(
     }
   }
   let value = AbstractValue.createAbstractObject(realm, propsName || "props");
+  realm.react.abstractHints.set(value, "HAS_NO_KEY_OR_REF");
   invariant(value instanceof AbstractObjectValue);
```

Comment:
While we’re at it could we also mark it as frozen?

Reply:
I just tried. Seems that we can't mark `AbstractObjectValue` as frozen :( Prepack only supports `ObjectValue`